presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -190,6 +190,29 @@ public void testFiltersAndProjections()
         assertQuery("SELECT if(orderkey % 2 = 0, quantity_by_linenumber) FROM orders_ex");
     }
 
+    @Test
+    public void testAnalyzeStats()
+    {
+        assertQuery("ANALYZE region");
+        assertQuery("SHOW STATS FOR region");
+        // Create a partitioned table and run analyze on it.
+        String tmpTableName = generateRandomTableName();
+        try {
+            getQueryRunner().execute(getSession(), String.format("CREATE TABLE %s (name VARCHAR, regionkey BIGINT," +
+                    "nationkey BIGINT) WITH (partitioned_by = ARRAY['regionkey','nationkey'])", tmpTableName));
+            Session writeSession = buildSessionForTableWrite();
+            getQueryRunner().execute(writeSession,
+                    String.format("INSERT INTO %s SELECT name, regionkey, nationkey FROM nation", tmpTableName));
+            assertQuery(String.format("SELECT * FROM %s", tmpTableName),
+                    "SELECT name, regionkey, nationkey FROM nation");
+            assertQuery("ANALYZE %s WITH (partitions = ARRAY[ARRAY['0','0'],ARRAY['4', '11']])", tmpTableName);
```

Comment:
Might be worth adding a comment about the results returned by ANALYZE for this query. Is one of the arrays an empty result ?

Reply:
I dont' think so. Here are the partitions would look like:

```
presto:tpch> select * from "tmpregion$partitions";
 regionkey | nationkey 
-----------+-----------
         2 |        12 
         2 |         8 
         2 |         9 
         2 |        18 
         2 |        21 
         4 |        13 
         4 |        11 
         4 |        20 
         4 |        10 
         4 |         4 
         3 |        23 
         3 |        22 
         3 |         6 
         3 |         7 
         3 |        19 
         1 |        24 
         1 |         1 
         1 |        17 
         1 |         2 
         1 |         3 
         0 |        15 
         0 |        14 
         0 |         0 
         0 |        16 
         0 |         5 
(25 rows)
```